app/models/mixins/archived_mixin.rb

```diff
@@ -2,11 +2,14 @@ module ArchivedMixin
   extend ActiveSupport::Concern
 
   included do
+    scope :deleted, -> { where(:deleted => true) }
+    scope :not_deleted, -> { where(:deleted => false) }
```

Comment:
ah, actually I think we should define relation for active and for all_, so we can have a proper dependent destroy